articles/101_build_tool.md

```diff
@@ -48,43 +54,74 @@ In the ROS ecosystem [rosdep](http://wiki.ros.org/rosdep) can be used for this.
 The build tool also does not create binary packages (e.g. a Debian package).
 In the ROS ecosystem [bloom](http://wiki.ros.org/bloom) is used to generate the required metadata and then platform dependent tools like `dpkg-buildpackage` build binary packages.
 
-## Build Tool vs. Build System
+## Build Tool vs. Build System vs. Package manager
+
+A build system schedules and executes atomic tasks for creating runnable software from sources.
+Task examples are test, compile, install.
+Examples are Gnu Make, cmake, python setuptools.
+
+A build tool schedules and invokes the build systems for separate source trees (packages) in topological order of dependency.
+Example are rosbuild, catkin_make, ament.
+
+A package manager downloads, optionally builds, then installs released packages from origin locations.
+Examples are dpkg, rpm, homebrew, portage, robotpkg.
+
+## Build tool features
+
+The following describes the essential functional requirements for a build tool, and further functional and non-functional requirements that are to be decided.
+
+### Topological invocation of build systems
 
-A build tool operates on a set of packages.
-It determines the dependency graph and invokes the package specific build tool of each package in topological order.
-The build tool itself should know as little as possible about the build system used for a specific package.
-Just enough in order to know how to setup the environment for it, invoke the build, and setup the environment to use the built package.
+A build tool defines a manifest format for packages to follow to define the invocation of the respective build tool, and the dependencies to other packages.
```

Comment:
Is it a goal *not* to depend on any fixed manifest format? Then that would belong in the goals section (or somewhere else in the document, anyway)